### ISSUE TITLE
crashes after choosing to download another

### ISSUE THREAD

**Issue by gitdunk** [Bijman/srb2dl#13]
(P.S I replaced all the home directories with the $HOME variable) When I finish selecting my first mod and hit yes to the "Do you want to download other stuff?" prompt, it crashes with `$HOME/.local/bin/srb2dl: line 5604: $HOME/.local/bin/srb2dl: Argument list too long
$HOME/.local/bin/srb2dl: line 5604: $HOME/.local/bin/srb2dl: Success`
disabling autodir didn't help, and neither did selecting a different mod to download first

I hand copied everything in my console from using SRB2DLDEBUG=1 to a text file and here it is (it got cut off because of the sheer amount of stuff that happened after I hit yes)
[srb2dl.log](https://github.com/user-attachments/files/16826171/srb2dl.log)

this is what I got when I output to a text file via > instead, they are completely different
```
[1;94m
Please choose category by typing number and then pressing enter. Press enter or ctrl+c to quit.
[0m1) Characters
2) Maps
3) Multi-Category
4) Gametypes
5) Assets
6) Lua
7) 3D Models
8) Player Colors
9) Miscellaneous
10) Ported Characters
11) Ported Maps
12) Ported Assets
13) Ported Lua
14) Ported 3D Models
15) Ported Miscellaneous
16) Software
17) Builds
18) [1m[34mDone[0m
[1;97m#? [0m[H[2J[3J[1;94m
Please choose resource by typing number and then pressing enter. Press enter or ctrl+c to quit.
[0m1) [7;34m[1m[Open Assets][0m[0m [1mKombi[0m - [2mThe Wario Land 4 at Home:[0m - [37mUser: IHateThoseKoopalings[0m - [0m[37mUpdated: Thursday at 3:19 AM[0m - [1;33m★ ★ ★ ★ ☆[0m
2) [1mBlossom the Raccoon (and Flimp)[0m - [2mA kind-hearted raccoon with winds of freedom.[0m - [37mUser: Spectra[0m - [0m[37mUpdated: Aug 24, 2024[0m - [1;33m★ ★ ★ ★ ☆[0m
3) [1mX Sonic[0m - [2mX Sonic speeds his way into SRB2![0m - [37mUser: BlueBlur[0m - [0m[37mUpdated: Aug 23, 2024[0m - [1;33m★ ★ ★ ★ ⯪[0m
4) [7;34m[1m[Open Assets][0m[0m [1mRafael (REMAKE!) [v5.1.2] - Shelved to oblivion[0m - [2mThe guy that just wants to chill around.[0m - [37mUser: cozyraf[0m - [0m[37mUpdated: Aug 22, 2024[0m - [1;33m★ ★ ★ ★ ☆[0m
5) [1mFlames The Hedgehog[0m - [2mYep Another Character Mod[0m - [37mUser: jimmysummon05[0m - [0m[37mUpdated: Aug 20, 2024[0m - [1;33m★ ☆ ☆ ☆ ☆[0m
6) [1mMach the Pangolin[0m - [2mThe world's fastest Pangolin![0m - [37mUser: ScarletMarx[0m - [0m[37mUpdated: Aug 18, 2024[0m - [1;33m★ ★ ★ ★ ☆[0m
7) [1mGemma the Tarsier-Cat[0m - [2mMeet the young Half-Primate, Half-Feline Earthshaker![0m - [37mUser: RalphJeremy65[0m - [0m[37mUpdated: Aug 14, 2024[0m - [1;33m★ ★ ★ ★ ⯪[0m
8) [1mScribble the What Now?[0m - [2mHe's not Sonic! He's Scribble![0m - [37mUser: GamerKid[0m - [0m[37mUpdated: Aug 9, 2024[0m - [1;33m★ ★ ★ ★ ☆[0m
9) [1mCacee "Kiwi" Cactus -v2.1.1[0m - [2mCacee is here to biff and baff![0m - [37mUser: light dasher[0m - [0m[37mUpdated: Aug 5, 2024[0m - [1;33m★ ★ ★ ★ ⯪[0m
10) [1mVeph the Dolphin-Fox[0m - [2mThe Energetic Two-Part Mammal Makes a Splash![0m - [37mUser: SMS Alfredo[0m - [0m[37mUpdated: Aug 5, 2024[0m - [1;33m★ ★ ★ ★ ⯪[0m
11) [1mDoomGuy[0m - [2mThe famous UAC marine arrives on SRB2![0m - [37mUser: Lugent[0m - [0m[37mUpdated: Aug 2, 2024[0m - [1;33m★ ★ ★ ★ ☆[0m
12) [1mTA Tails Revamped[0m - [2mStraight Outta Cocoa Island[0m - [37mUser: WhisperVII[0m - [0m[37mUpdated: Jul 29, 2024[0m - [1;33m★ ★ ★ ★ ☆[0m
13) [7;34m[1m[Open Assets][0m[0m [1mSeles The Arctic Fox[0m - [2mHe lives in Frozen Hillside Zone.[0m - [37mUser: Sls64LGamingBro[0m - [0m[37mUpdated: Jul 28, 2024[0m - [1;33m★ ★ ★ ⯪ ☆[0m
14) [1mMint "the AmyDroid" Sweetnam[0m - [2mA pretty fast character[0m - [37mUser: SSNeoMint[0m - [0m[37mUpdated: Jul 26, 2024[0m - [1;33m★ ★ ★ ★ ☆[0m
15) [1mDark94 & XKZ[0m - [2mIt's me! And other person too![0m - [37mUser: Dark94[0m - [0m[37mUpdated: Jul 25, 2024[0m - [1;33m★ ★ ★ ★ ☆[0m
16) [7;34m[1m[Open Assets][0m[0m [1mUgly[0m - [2mNow in Technicolor![0m - [37mUser: Average[0m - [0m[37mUpdated: Jul 25, 2024[0m - [1;33m★ ★ ★ ★ ⯪[0m
17) [7;34m[1m[Open Assets][0m[0m [1mHatsune Miku[0m - [2mIdol From Vocaloid[0m - [37mUser: Shanon[0m - [0m[37mUpdated: Jul 25, 2024[0m - [1;33m★ ★ ★ ★ ★[0m
18) [1mStephChars v3 - Bold & Brash Update[0m - [2mTrio of Adventure, Speed and Destruction![0m - [37mUser: DrStephen[0m - [0m[37mUpdated: Jul 23, 2024[0m - [1;33m★ ★ ★ ★ ⯪[0m
19) [1mLegend of Heroes Trails Character Pack[0m - [2mA pack with Legend of Heroes Trails characters for SRB2 Persona[0m - [37mUser: HajimeBannings[0m - [0m[37mUpdated: Jul 20, 2024[0m - [1;33m★ ★ ★ ★ ☆[0m
20) [7;34m[1m[Open Assets][0m[0m [1mSpongebob Squarepants: Supersponge[0m - [2mInspired by his heroes, Mermaidman and Barnacle Boy, Spongebob embarks on a big adventure![0m - [37mUser: Globx[0m - [0m[37mUpdated: Jul 14, 2024[0m - [1;33m★ ★ ★ ★ ★[0m
21) [1m[34mPrevious Page 0[0m
22) [1m[34mNext Page 2[0m
23) [1m[34mBack to categories[0m
24) [1m[34mDone[0m
[1;97m#? [0m[H[2J[3J[1;94m
Please choose file by typing number and then pressing enter. Press enter or ctrl+c to quit.
[0m1) CL_StephCharsOC-v3.pk3
2) CL_Skip-v1.pk3
3) [1m[34mBack to addons[0m
4) [1m[34mBack to categories[0m
5) [1m[34mHistory[0m
6) [1m[34mDone[0m
[1;97m#? [0m[H[2J[3J[1;97mFilelist:
CL_StephCharsOC-v3.pk3[0m[1;97m

Directory:
[0m[1;94m

Please choose directory to download resource by typing number and then pressing enter. Press enter or ctrl+c to quit.
[0m1) addons
2) addons/character
3) addons/maps
4) logs
[1;97m#? [0m[H[2J[3J[1;97mFilelist:
CL_StephCharsOC-v3.pk3[0m[1;97m

Directory:

$HOME/.var/app/org.srb2.SRB2/.srb2/addons/character[0m[1;94m #replaced home directory with variable for privacy

Do you want to download other stuff?
[0m1) Yes
2) No
[1;97m#? [0m`
```

**Comment by Bijman** [comment]
Thank you for reporting and log. This is caused by limit length for arguments in command. In this situation it is stored HTML data of SRB2 MB page to script's variable PAGECODE. I tried with "ulimit -s unlimited" method, but unfortunately no dice. Another way would be storing temporarily HTML data to file in path /tmp and redirect it to command..

**Comment by Bijman** [comment]
Fixed with commit 3757665. I removed variable PAGECODE  storing HTML data, which causes issue, from some commands as it didn't have any usage.
